src/charm.py

```diff
@@ -230,32 +246,51 @@ def self_scraping_job(self):
 
         return [job]
 
+    def _update_blackbox_config_yaml_given_dict_from_relation(self, modules: dict) -> None:
+        """Update the blackbox config yaml given a dict of modules defined in relation.
+
+        This function takes the dict of modules from the BlackboxExporterRequirer and
+        updates the config file with the required modules, only if they do not exist already
+        in the config file.
+
+        Raises:
+            yaml.YAMLError: If there is an error in the YAML formatting or parsing.
+            TypeError: If type is not a string, as `yaml.safe_load` requires a string input.
+        """
+        if not modules:
+            return
+
+        config_file_data = self.container.pull(self._config_path).read()
```

Comment:
This method is called by the constructor. Do we need a `can_connect` guard somewhere?

Reply:
added `can_connect` to safeguard this function in case it gets called from somewhere else in the future, in 734f467b0b1333a705578a954fc0eb14f3b4e04b